airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py

```diff
@@ -161,8 +160,13 @@ def manage_template_specs(self):
         return template_body
 
     def create_job_name(self):
-        initial_name = add_unique_suffix(name=self.task_id, max_len=MAX_LABEL_LEN)
-        return re.sub(r"[^a-z0-9-]+", "-", initial_name.lower())
+        name = (
+            self.name or self.template_body.get("spark", {}).get("metadata", {}).get("name") or self.task_id
+        )
+
+        updated_name = add_unique_suffix(name=name, max_len=MAX_LABEL_LEN)
+
+        return re.sub(r"[^a-z0-9-]+", "-", updated_name.lower())
```

Comment:
maybe better use this function here?
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/pod.py#L1007

Reply:
Yes make sense 😄  updated now.